chalice/package.py

```diff
@@ -357,15 +386,255 @@ def _register_cfn_resource_name(self, name):
+            resource.resource_name] = {
+                'statement_id': resource.resource_name,
+                'action': 'lambda:InvokeFunction',
+                'function_name': self._fref(resource.lambda_function),
+                'principal': 's3.amazonaws.com',
+                'source_arn': 'arn:aws:s3:::%s' % resource.bucket
+        }
+
+    def _generate_sqseventsource(self, resource, template):
+        # type: (models.SQSEventSource, Dict[str, Any]) -> None
+        template['resource'].setdefault('aws_lambda_event_source_mapping', {})[
+            resource.resource_name] = {
+                'event_source_arn': self._arnref(
+                    "arn:aws:sqs:%(region)s:%(account_id)s:%(queue)s",
+                    queue=resource.queue),
+                'batch_size': resource.batch_size,
+                'function_name': self._fref(resource.lambda_function),
+        }
+
+    def _generate_snslambdasubscription(self, resource, template):
+        # type: (models.SNSLambdaSubscription, Dict[str, Any]) -> None
+
+        if resource.topic.startswith('arn:aws'):
+            topic_arn = resource.topic
+        else:
+            topic_arn = self._arnref(
+                'arn:aws:sns:%(region)s:%(account_id)s:%(topic)s',
+                topic=resource.topic)
+
+        template['resource'].setdefault('aws_sns_topic_subscription', {})[
+            resource.resource_name] = {
+                'topic_arn': topic_arn,
+                'protocol': 'lambda',
+                'endpoint': self._fref(resource.lambda_function)
+        }
+        template['resource'].setdefault('aws_lambda_permission', {})[
+            resource.resource_name] = {
+                'function_name': self._fref(
+                    resource.lambda_function),
+                'action': 'lambda:InvokeFunction',
+                'principal': 'sns.amazonaws.com',
+                'source_arn': topic_arn
+        }
+
+    def _generate_scheduledevent(self, resource, template):
+        # type: (models.ScheduledEvent, Dict[str, Any]) -> None
+
+        template['resource'].setdefault(
+            'aws_cloudwatch_event_rule', {})[
+                resource.resource_name] = {
+                    'name': resource.resource_name,
+                    'schedule_expression': resource.schedule_expression
+        }
+        template['resource'].setdefault(
+            'aws_cloudwatch_event_target', {})[
+                resource.resource_name] = {
+                    'rule': '${aws_cloudwatch_event_rule.%s.name}' % (
+                        resource.resource_name),
+                    'target_id': resource.resource_name,
+                    'arn': self._fref(resource.lambda_function)
+        }
+        template['resource'].setdefault(
+            'aws_lambda_permission', {})[
+                resource.resource_name] = {
+                    'function_name': self._fref(resource.lambda_function),
```

Comment:
So I noticed this in the generate terraform template, but for the function name we actually end up referencing the `arn`. Using the arn seemed to be fine, but may be worth double checking if we should be using the name instead.

Reply:
I switched this to name but might  switch back, its actually bit easier on testing with the arn ref as there isn't really a name for the app setup in most tests, so it comes out oddly when doing assertions as while the sample apps have a name the configs for them don't so assertions come out as comparisons to None-function-name

Reply:
left with function name unless parameter name said arn, and updated said tests to generate config with chalice app name for clean comparisons.